R/probability_distributions.R

```diff
@@ -9,11 +9,12 @@ uniform_distribution <- R6Class (
 
     initialize = function (min, max, dim) {
 
+      dim <- check_dims(min, max, target_dim = dim)
```

Comment:
Since `check_dims()` is expecting greta arrays, I think you might need to explicitly coerce `min` and `max` to have two dimensions before doing this. What happens when you do: `uniform(1:3, 4)` with this version? I would expect it to error.